### DISCUSSION TITLE
Moving from streamlit to budibase

### DISCUSSION THREAD

**Discussion by benyi1981** [Budibase/budibase#12952] [Help]
Hi, I'm a complete newbie and need help.

I wrote a machine learning app with streamlit. I'd like to move to Budibase because it looks more professional. What's the best way to set this up?

The machine learning code is all written in python, and I have a docker container over the entire streamlit app. I'd like to re-use as much of the python code as possible. The python code extracts and processes data, creates machine learning model, runs inferences. I could turn them into APIs with fast_api. 

The python code currently also does some chart visualization, which I intend to replace with Budibase.

I'm thinking keeping Budibase and python backend separate, and interface between them via APIs. Is this the best approach?

**Comment by ConorWebb96** [comment]
Hey @benyi1981,

I would definitely recommend doing this through an API. I don't believe there is any other way. You should be able to easily interact with this using the [rest data source](https://docs.budibase.com/docs/rest). We have quite a bit of documentation on this. You can import a collection, or add your rest api requests one at a time its completely up to you.

Bear in mind if you do end up going the api route and are wanting to make this secure. Make sure that you either go the bearer token approach or the basic auth approach as I don't believe there is support for other authentication methods.

I hope this information helps you with making your decision.
